packages/api/src/controllers/clip.ts

```diff
@@ -0,0 +1,154 @@
+import { validatePost } from "../middleware";
+import { Request, Router } from "express";
+import _ from "lodash";
+import { db } from "../store";
+import { NotFoundError } from "../store/errors";
+import { pathJoin } from "../controllers/helpers";
+import {
+  createAsset,
+  validateAssetPayload,
+  defaultObjectStoreId,
+  catalystPipelineStrategy,
+} from "./asset";
+import { generateUniquePlaybackId } from "./generate-keys";
+import { v4 as uuid } from "uuid";
+import { DBSession } from "../store/db";
+import { fetchWithTimeout } from "../util";
+import { DBStream } from "../store/stream-table";
+
+const app = Router();
+
+app.post("/", validatePost("clip-payload"), async (req, res) => {
+  const playbackId = req.body.playbackId;
+  const userId = req.user.id;
+
+  const id = uuid();
+  let uPlaybackId = await generateUniquePlaybackId(id);
+
+  const content = await db.stream.getByPlaybackId(playbackId); //||
```

Comment:
Shouldn't this be reading from the session table instead? Conceptually, the clips are made on the session, not the channel (parent stream, being fetched here). 
e.g. A time range like "01:23 to 03:46" doesn't really make sense when it references a channel instead of a specific session (unless we're using absolute timestamps, are we?)

edit: Ok, I see that you fetch the session below, but the problem is that the clip asset is referencing the stream but not the session. Might be the case to just move the `validateAssetPayload` to after fetching the session ~L84, so we can create the asset payload with the proper reference to the session instead (from the session we can always find the channel, but not the other way around)

Reply:
While processing, the clip is going to reside in the recordings storage under playbackId/sessionId/clip_assetPlaybackId/clip.m3u8 so the original playbackId of the content is needed to build this URL down the pipeline - although there is room for improvement in task runner and there are some hacks in catalyst-api that are going to get resolved and have a cleaner task here in studio.

Also, now the asset is going to use the normal vod storage for the output location generation, so at the end the output asset won't have the sessionId in the output URL - will need to put it in the source object to reference clips to sessions

Reply:
I think this is independent of where the files are stored. On the API level, the clips should reference the user-accessible content where the clip was based from. In case of a livestream, the clip is not made on top of a "channel", but on top of a specific session, so this is what has to be referenced on the asset side. 

> While processing, the clip is going to reside in the recordings storage under playbackId/sessionId/clip_assetPlaybackId/clip.m3u8

Not sure what you mean by "recording storage", but isn't the clip just gonna be a regular asset stored under `/playbackId/index.m3u8` (the asset `playbackId`, not the original one)? Feels like you're referring to an intermediate object created during catalyst processing, which doesn't really matter here.

If you need the `playbackId` to be able to find the temporary recording of the livestream and create the clip from it, then you're right, you will need the `playbackId` here as well which isn't available on the `session` object. That would be just another output from the function I suggested though, returning both the `playbackId` and the `Asset | Session`.